tensorflow/compiler/xla/service/gpu/gpu_compiler.cc

```diff
@@ -826,9 +826,9 @@ Status GpuCompiler::OptimizeHloPostLayoutAssignment(
     });
     pipeline.AddPass<HloPassFix<MoveCopyToUsers>>();
 
+#if GOOGLE_CUDA
```

Comment:
Could we avoid CUDA guards here? What is the desired behavior change?

Reply:
Thanks for the feedback! 

Yes, I have replaced all related passes that used to broken rocm build (e.g., gemm_rewriter, gemm_rewriter_triton and tree_reduction) with GpuVersion as I did it in above, and tested ok on rocm and cuda node.  No more macro and it is more generic.  :) 

But I am not 100% sure about gemm_rewriter_test on fp8 ones as my cuda-node cannot pass it even with a clean repo, but I think my changes wouldn't affect these parts. Please let me know if it needs any changes.

Reply:
@cheshire gemm_rewriter_triton is updating quickly, so I have to change some of code daily to avoid the conflict. 

Wonder will you be able to review this PR recently? or maybe shall I switch this PR to openXLA?

Reply:
> gemm_rewriter_triton is updating quickly

Probably better to split up this PR then?

> Wonder will you be able to review this PR recently? or maybe shall I switch this PR to openXLA?

Switching is a good idea, but splitting would also make the review simpler.